doc/adr/0018-synchronising-queries-across-indexers-running-in-parallel.rst

```diff
@@ -0,0 +1,51 @@
+
+This solution also forgoes the possibility of splitting up the indexers into different processes or machines. Our customers complained about the excessive resource consumption, performance and the inability to easily add their own indexers to the chain index. Giving users the possibility of distributing their indexers will partially address the first concern by splitting resource consumption on several machines (which is also addressed at the indexer design level by an inherently scalable design) and indirectly the second (since indexers are distributable, they have a simple, standard API that allows any customer to roll their own version, implicitly solving the hard problems like rollbacks and synchronisation).
+
+Decision
+--------
+
+1. Change the `Request` type to include a preferred slot number and how many results we want to receive.
+2. Change the `Response` type to include all the requested query results encoded as a base query result followed by other responses.
+3. Add a new function to the `Queryable` type class that witnesses the folding of Events into the query result.
+
+Argument
+--------
+
+A fairly simple solution to the synchronisation problem would be to qualify responses by the current slot number. This is a reasonable first thought, but it does have a high probability of returning results that are qualified by different slot numbers (if the indexers are fully synchronised, it would be a small offset, but still dependent on the various indexers processing speeds). The advantage of this solution is that at least, we now know that the indexers were not fully synchronised. If the application does not require them to be, then this is a reasonable solution.
+
+If, however, the application requires all queries to run at the same slot number, this solution will simply not function. What we could do is request multiple query responses from indexers in the hope that we can find a common slot number where all indexers have an answer. Is this hope realistic? I would argue that it is realistic for indexers that have been fully synchronised with the blockchain since the processed slot offset between them should be fairly small.
+
+Now that will solve our functional problem, but it may introduce some non-functional ones. First, the query results may be fairly big, and returning more of them will put pressure on the network and the parsing of data by the client applications. This can be partially mitigated by selecting a smaller interval, but maybe we can help out here by improving the design a bit. One observation is that we don't actually need to send the full query results. We need to send a query result for the oldest slot number available, but for the newer ones, we only need to send the difference recorded by the next slot. This implementation maps very well to the way we store information in the indexer (an accumulator and a set of events).
```

Comment:
Should this paragraph be under the `Implications (or consequences)` of the[ ADR-template](https://github.com/input-output-hk/plutus-apps/blob/main/doc/adr/0001-record-architecture-decisions.rst#adr-template)

Reply:
I think this still describes the solution and the implementation. I am not sure why this would be classified as an implication. Can you, please, elaborate?

Reply:
Actually I think I understand what you meant. I added an `Implications` heading after this line.